compose: TestComposeCompare failed

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1904548&tab=buildLog) on [master@c3bb2bce3d69d0ceef20797d856a6ac107a3ef47](https://github.com/cockroachdb/cockroach/commits/c3bb2bce3d69d0ceef20797d856a6ac107a3ef47):

```
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_4, col1_9 ASC, col1_1 DESC, col1_10 DESC, col1_8) INCLUDE (col1_5, col1_6, col1_11);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 DESC, col1_4 DESC, col1_2, col1_3 DESC, col1_6 ASC, col1_7, col1_5 DESC, col1_1 ASC, col1_9, col1_8 DESC, col1_11 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_10 DESC, col1_7 DESC, col1_0 ASC, col1_6, col1_2 ASC, col1_9 ASC, col1_3 ASC, col1_11 ASC, col1_1 DESC, col1_5 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_6 ASC, col1_10 ASC, col1_5, col1_1 DESC, col1_3 ASC, col1_11 DESC, col1_2 ASC, col1_9, col1_0 ASC, col1_7, col1_8 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0, col1_9);
        test_1        |             CREATE TABLE table2 (col1_3 CHAR NOT NULL, col2_1 NAME NOT NULL, col2_2 INT8, col2_3 TIMESTAMPTZ, col2_4 INT8, col2_5 TEXT NULL, col2_6 TIME NULL, col2_7 OID NULL, col2_8 DECIMAL NULL, col2_9 BOOL, PRIMARY KEY (col1_3, col2_8, col2_2, col2_9, col2_6));
        test_1        |             CREATE INDEX ON table2 (col2_2 ASC) INCLUDE (col2_1, col2_5);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_6 ASC, col2_1 DESC, col2_8 ASC, col1_3);
        test_1        |             CREATE INDEX ON table2 (col2_6, col2_8) INCLUDE (col2_1, col2_3, col2_5);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_5 DESC, col2_4 DESC, col2_1 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_8 DESC, col2_6 DESC, col2_1, col2_2 DESC, col2_3 ASC, col2_9, col2_7 DESC) INCLUDE (col2_5);
        test_1        |             CREATE INDEX ON table2 (col2_4, col2_6 ASC, col2_2, col2_5 ASC, col2_8, col2_7);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_9, col2_6, col1_3 DESC, col2_3, col2_7 DESC, col2_1, col2_8, col2_2, col2_5 DESC);
        test_1        |             CREATE INDEX ON table2 (col2_3 ASC, col1_3 ASC, col2_7 DESC, col2_1 ASC, col2_6 DESC, col2_5 DESC, col2_8 ASC, col2_4 DESC, col2_9 ASC, col2_2 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_2);
        test_1        |             CREATE TABLE table3 (col1_3 CHAR NOT NULL, col2_8 DECIMAL NULL, col2_2 INT8, col2_9 BOOL, col2_6 TIME NULL, col3_5 BIT(14), col3_6 "char", col3_7 NAME NOT NULL, col3_8 NAME NOT NULL, PRIMARY KEY (col1_3, col2_8, col2_2, col2_9, col2_6, col3_8, col3_5, col3_6));
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_8 ASC, col3_5 ASC, col1_3 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col1_3, col2_8 ASC, col3_6, col3_8 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_5 DESC, col2_8, col3_7 DESC, col3_8 ASC, col2_9 ASC, col2_6 DESC, col2_2 ASC);
        test_1        |             CREATE INDEX ON table3 (col2_8, col1_3, col2_9, col3_7 ASC, col3_6, col2_6 DESC, col3_5 DESC, col3_8 ASC, col2_2);
        test_1        |             CREATE INDEX ON table3 (col2_8) INCLUDE (col3_7);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_5 DESC, col3_8, col2_9 DESC, col3_6, col3_7, col2_6 DESC, col1_3, col2_2, col2_8);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col2_9, col1_3 DESC, col3_6, col3_5, col3_7 DESC, col2_6 DESC, col2_8);
        test_1        |             CREATE TABLE table4 (col4_0 REGCLASS, col4_1 INT8 NULL, col4_2 OID NOT NULL, col4_3 TEXT, col4_4 TIMESTAMPTZ NULL, col4_5 TIMESTAMPTZ NULL, PRIMARY KEY (col4_2));
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 DESC, col4_4 DESC, col4_2, col4_3 ASC) INCLUDE (col4_0);
        test_1        |             CREATE INDEX ON table4 (col4_3 DESC, col4_1 ASC, col4_2);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_4, col4_0, col4_5 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_2 DESC, col4_0, col4_1 DESC, col4_5 DESC, col4_4 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_5, col4_0 DESC, col4_2 ASC, col4_3 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_4, col4_0) INCLUDE (col4_1, col4_3);
        test_1        |             CREATE INDEX ON table4 (col4_4 DESC, col4_1 DESC, col4_5, col4_2 DESC, col4_0 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_0, col4_4 DESC, col4_1, col4_2, col4_3 ASC, col4_5);
        test_1        |             CREATE TABLE table5 (col5_0 INTERVAL NOT NULL, col5_1 INT8, col5_2 "char" NOT NULL, col5_3 REGTYPE NULL, col5_4 REGPROC NULL, col5_5 TIME NULL, col5_6 REGCLASS NOT NULL, col5_7 REGPROC NOT NULL, col5_8 BYTEA, col5_9 TIMESTAMP NOT NULL, col5_10 "char", PRIMARY KEY (col5_2));
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_2);
        test_1        |             ALTER TABLE table4 ADD FOREIGN KEY (col4_1) REFERENCES table2 (col2_2);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col1_3, col2_3) REFERENCES table1 (col1_0, col1_9);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col2_5) REFERENCES table5 (col5_2) ON UPDATE CASCADE;
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: foreign key constraint "table2_col1_3_fkey" cannot be implemented (SQLSTATE 42804)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1925418&tab=buildLog) on [master@20916a30cf9356683c973f8653e8b69613a75fe4](https://github.com/cockroachdb/cockroach/commits/20916a30cf9356683c973f8653e8b69613a75fe4):

```
        postgres_1    | 	CREATE INDEX ON table2 (col2_8 ASC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_9 DESC, col2_13 DESC, col2_5 DESC, col2_2, col2_4, col2_12, col2_0 ASC, col2_6, col2_11 ASC, col2_3 ASC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_2, col2_6 ASC, col2_8, col2_0 ASC, col2_9 DESC, col2_12 ASC, col2_4, col2_13 ASC, col2_14, col2_5, col2_3);
        postgres_1    | 	CREATE TABLE table3 (col1_5 FLOAT8 NOT NULL, col3_1 NAME, col3_2 REGPROCEDURE NOT NULL, col3_3 TIMETZ NULL, col3_4 DATE NULL, col3_5 REGTYPE, col3_6 REGCLASS NOT NULL, PRIMARY KEY (col1_5));
        postgres_1    | 	CREATE TABLE table4 (col1_5 FLOAT8 NOT NULL, col4_1 OID, col4_2 REGTYPE, col4_3 REGTYPE NOT NULL, col4_4 INET NOT NULL, PRIMARY KEY (col1_5, col4_2, col4_3, col4_1));
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_3 ASC, col4_4 ASC, col1_5 ASC);
        postgres_1    | 	CREATE INDEX ON table4 (col4_1 DESC, col4_4 ASC, col4_3);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_4 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_1, col4_2);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_2 DESC, col1_5 ASC);
        postgres_1    | 	CREATE INDEX ON table4 (col4_2 DESC, col4_3) INCLUDE (col4_4);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_4, col1_5 ASC, col4_2 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_1 ASC, col4_2, col1_5 ASC);
        postgres_1    | 	ALTER TABLE table3 ADD FOREIGN KEY (col3_1) REFERENCES table1 (col1_1) ON UPDATE RESTRICT;
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (603.97s)
        test_1        |     --- FAIL: TestCompare/postgres (3.22s)
        test_1        |         compare_test.go:128: CREATE TABLE table1 (col1_0 INT8 NOT NULL, col1_1 CHAR, col1_2 VARCHAR NOT NULL, col1_3 BOOL NOT NULL, col1_4 REGPROCEDURE NULL, col1_5 FLOAT8 NOT NULL, col1_6 DECIMAL NULL, col1_7 REGCLASS NOT NULL, col1_8 TIMESTAMPTZ NULL, col1_9 DATE, col1_10 TIMETZ, col1_11 INET NULL, col1_12 OID, col1_13 REGTYPE NULL, PRIMARY KEY (col1_5));
        test_1        |             CREATE INDEX ON table1 (col1_1 DESC, col1_9, col1_0, col1_13 ASC, col1_3 DESC, col1_6 ASC, col1_12, col1_5, col1_4 DESC, col1_7 DESC) INCLUDE (col1_10, col1_11);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_12, col1_8 DESC, col1_6, col1_5 DESC, col1_0 DESC, col1_13 ASC, col1_3 DESC, col1_2, col1_4 ASC, col1_10 ASC) INCLUDE (col1_1, col1_9, col1_11);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_1);
        test_1        |             CREATE TABLE table2 (col2_0 INET, col2_1 TIMESTAMPTZ, col2_2 TIMETZ NULL, col2_3 INTERVAL NULL, col2_4 BIT(46), col2_5 TIME, col2_6 DECIMAL NOT NULL, col2_7 DECIMAL NOT NULL, col2_8 TIMESTAMP, col2_9 UUID NOT NULL, col2_10 OID NOT NULL, col2_11 REGTYPE NOT NULL, col2_12 UUID NOT NULL, col2_13 NAME NULL, col2_14 VARCHAR NULL, PRIMARY KEY (col2_10));
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_10 DESC, col2_8 ASC, col2_4, col2_3 ASC, col2_14 ASC, col2_12, col2_0, col2_6 DESC, col2_5 DESC) INCLUDE (col2_2, col2_7, col2_9, col2_13);
        test_1        |             CREATE INDEX ON table2 (col2_8 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_9 DESC, col2_13 DESC, col2_5 DESC, col2_2, col2_4, col2_12, col2_0 ASC, col2_6, col2_11 ASC, col2_3 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_2, col2_6 ASC, col2_8, col2_0 ASC, col2_9 DESC, col2_12 ASC, col2_4, col2_13 ASC, col2_14, col2_5, col2_3);
        test_1        |             CREATE TABLE table3 (col1_5 FLOAT8 NOT NULL, col3_1 NAME, col3_2 REGPROCEDURE NOT NULL, col3_3 TIMETZ NULL, col3_4 DATE NULL, col3_5 REGTYPE, col3_6 REGCLASS NOT NULL, PRIMARY KEY (col1_5));
        test_1        |             CREATE TABLE table4 (col1_5 FLOAT8 NOT NULL, col4_1 OID, col4_2 REGTYPE, col4_3 REGTYPE NOT NULL, col4_4 INET NOT NULL, PRIMARY KEY (col1_5, col4_2, col4_3, col4_1));
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_3 ASC, col4_4 ASC, col1_5 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_1 DESC, col4_4 ASC, col4_3);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_4 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1, col4_2);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 DESC, col1_5 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_2 DESC, col4_3) INCLUDE (col4_4);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_4, col1_5 ASC, col4_2 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 ASC, col4_2, col1_5 ASC);
        test_1        |             ALTER TABLE table3 ADD FOREIGN KEY (col3_1) REFERENCES table1 (col1_1) ON UPDATE RESTRICT;
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: foreign key constraint "table3_col3_1_fkey" cannot be implemented (SQLSTATE 42804)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1934025&tab=buildLog) on [master@e6b47088aba9c9978501b966a0e88aeb273d9990](https://github.com/cockroachdb/cockroach/commits/e6b47088aba9c9978501b966a0e88aeb273d9990):

```
        postgres_1    | 				tab_119456.col1_1 AS col_155765,
        postgres_1    | 				NULL AS col_155766
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table1 AS tab_119456
        postgres_1    | 			WHERE
        postgres_1    | 				true
        postgres_1    | 		)
        postgres_1    | 			AS tab_119457
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_155763 NULLS FIRST, col_155764 NULLS FIRST, col_155765 NULLS FIRST, col_155766 NULLS FIRST
        test_1        | --- FAIL: TestCompare (731.84s)
        test_1        |     --- FAIL: TestCompare/postgres (130.01s)
        test_1        |         compare_test.go:154: compare cockroach1 to postgres:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |               	&pgtype.NumericArray{
        test_1        |               		Elements: []pgtype.Numeric{
        test_1        |               			{
        test_1        |             - 				Int:    s"5",
        test_1        |             + 				Int:    s"50000000000000000",
        test_1        |             - 				Exp:    0,
        test_1        |             + 				Exp:    -16,
        test_1        |               				Status: 2,
        test_1        |               			},
        test_1        |               		},
        test_1        |               		Dimensions: []pgtype.ArrayDimension{{Length: 1, LowerBound: 1}},
        test_1        |               		Status:     2,
        test_1        |               	},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT ((5::INT8::INT8 ^ 1::DECIMAL::DECIMAL)::DECIMAL::DECIMAL || NULL::DECIMAL[])::DECIMAL[] AS col_155868)
        test_1        |             		AS tab_119537
        test_1        |             ORDER BY
        test_1        |             	col_155868 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1948996&tab=buildLog) on [master@2cbf620cd229a55622fe17fb15d20ada1dbcccd3](https://github.com/cockroachdb/cockroach/commits/2cbf620cd229a55622fe17fb15d20ada1dbcccd3):

```
        postgres_1    | 				NULL::GEOMETRY NOT IN (NULL, NULL)
        postgres_1    | 		)
        postgres_1    | 			AS tab_31437
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_57523 NULLS FIRST, col_57524 NULLS FIRST
        postgres_1    | 2020-05-17 05:20:39.638 UTC [65] ERROR:  type "geometry" does not exist at character 134
        postgres_1    | 2020-05-17 05:20:39.638 UTC [65] STATEMENT:  SELECT
        postgres_1    | 		*
        postgres_1    | 	FROM
        postgres_1    | 		(
        postgres_1    | 			SELECT
        postgres_1    | 				tab_31484.col1_4 AS col_57591
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table1 AS tab_31484
        postgres_1    | 			WHERE
        postgres_1    | 				st_covers(NULL::GEOMETRY, NULL::GEOMETRY)::BOOL
        postgres_1    | 			GROUP BY
        postgres_1    | 				tab_31484.col1_4
        postgres_1    | 		)
        postgres_1    | 			AS tab_31485
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_57591 NULLS FIRST
        test_1        | --- FAIL: TestCompare (645.73s)
        test_1        |     --- FAIL: TestCompare/postgres (44.82s)
        test_1        |         compare_test.go:154: compare postgres to cockroach1:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	s"1",
        test_1        |             + 	s"0",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT floor((0::DECIMAL::DECIMAL ^ 0::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL AS col_57706) AS tab_31540
        test_1        |             ORDER BY
        test_1        |             	col_57706 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1950203&tab=buildLog) on [master@7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f](https://github.com/cockroachdb/cockroach/commits/7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f):

```
        postgres_1    | 				tab_23611.col2_13 AS col_42651,
        postgres_1    | 				NULL AS col_42652
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table2 AS tab_23611
        postgres_1    | 			WHERE
        postgres_1    | 				(NULL OR st_intersects(NULL::GEOGRAPHY(GEOMETRY,4326), NULL::GEOGRAPHY(GEOMETRY,4326))::BOOL)
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab_23611.col2_11
        postgres_1    | 		)
        postgres_1    | 			AS tab_23612
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_42647 NULLS FIRST,
        postgres_1    | 		col_42648 NULLS FIRST,
        postgres_1    | 		col_42649 NULLS FIRST,
        postgres_1    | 		col_42650 NULLS FIRST,
        postgres_1    | 		col_42651 NULLS FIRST,
        postgres_1    | 		col_42652 NULLS FIRST
        test_1        | --- FAIL: TestCompare (642.27s)
        test_1        |     --- FAIL: TestCompare/postgres (40.36s)
        test_1        |         compare_test.go:154: compare cockroach1 to postgres:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	s"0",
        test_1        |             + 	s"4",
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			floor((pow(0::DECIMAL::DECIMAL, mod(9::DECIMAL::DECIMAL, 9::DECIMAL::DECIMAL)::DECIMAL::DECIMAL)::DECIMAL::DECIMAL * 4::INT8::INT8)::DECIMAL::DECIMAL)::DECIMAL
        test_1        |             				AS col_42925
        test_1        |             	)
        test_1        |             		AS tab_23754
        test_1        |             ORDER BY
        test_1        |             	col_42925 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1962548&tab=buildLog) on [master@d5dd67bc846b73cebe8dc73b320de045ce8a170c](https://github.com/cockroachdb/cockroach/commits/d5dd67bc846b73cebe8dc73b320de045ce8a170c):

```
        test_1        | --- FAIL: TestCompare (1048.95s)
        test_1        |     --- FAIL: TestCompare/mutators (446.91s)
        test_1        |         compare_test.go:160: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		WITH
        test_1        |             			with_51300 (col_301983)
        test_1        |             				AS (
        test_1        |             					SELECT
        test_1        |             						*
        test_1        |             					FROM
        test_1        |             						(
        test_1        |             							VALUES
        test_1        |             								(COALESCE(NULL, '0105000020E610000000000000'::GEOGRAPHY)),
        test_1        |             								('0102000020E610000000000000'::GEOGRAPHY),
        test_1        |             								('0101000020E6100000000000000000F03F000000000000F03F'::GEOGRAPHY),
        test_1        |             								('0101000020E6100000000000000000F03F000000000000F03F'::GEOGRAPHY),
        test_1        |             								(
        test_1        |             									st_segmentize('0106000020E61000000200000001030000000100000004000000000000000000444000000000000044400000000000003440000000000080464000000000008046400000000000003E4000000000000044400000000000004440010300000002000000060000000000000000003440000000000080414000000000000024400000000000003E40000000000000244000000000000024400000000000003E4000000000000014400000000000804640000000000000344000000000000034400000000000804140040000000000000000003E40000000000000344000000000000034400000000000002E40000000000000344000000000000039400000000000003E400000000000003440'::GEOGRAPHY::GEOGRAPHY, 0.08593595658355241::FLOAT8::FLOAT8)::GEOGRAPHY
        test_1        |             								)
        test_1        |             						)
        test_1        |             							AS tab_192156 (col_301983)
        test_1        |             				),
        test_1        |             			with_51301 (col_301984)
        test_1        |             				AS (
        test_1        |             					SELECT
        test_1        |             						*
        test_1        |             					FROM
        test_1        |             						(VALUES (e'\x00'::TEXT), ('lt'::TEXT), (NULL), ('BfF_ZJFN'::TEXT))
        test_1        |             							AS tab_192157 (col_301984)
        test_1        |             				)
        test_1        |             		SELECT
        test_1        |             			'e]/'::TEXT::NAME AS col_301985
        test_1        |             	)
        test_1        |             		AS tab_192158
        test_1        |             ORDER BY
        test_1        |             	col_301985 NULLS FIRST
        test_1        |         compare_test.go:161: cockroach2: ping: context deadline exceeded
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2044002&tab=buildLog) on [master@8f768ad14cfb3f514db6d40465b2dd60ee1f2890](https://github.com/cockroachdb/cockroach/commits/8f768ad14cfb3f514db6d40465b2dd60ee1f2890):

```
        postgres_1    | 2020-06-27 05:13:17.098 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2020-06-27 05:13:17.118 UTC [56] LOG:  database system was shut down at 2020-06-27 05:13:17 UTC
        postgres_1    | 2020-06-27 05:13:17.125 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2020-06-27 05:13:18.889 UTC [64] ERROR:  unrecognized configuration parameter "experimental_enable_enums"
        postgres_1    | 2020-06-27 05:13:18.889 UTC [64] STATEMENT:  SET experimental_enable_enums = true;
        postgres_1    | 	CREATE TABLE table1 (col1_0 REGPROCEDURE NOT NULL, col1_1 BIT(32) NOT NULL, col1_2 INTERVAL NOT NULL, col1_3 DATE NOT NULL, col1_4 FLOAT8 NOT NULL, col1_5 OID NOT NULL, col1_6 FLOAT8 NOT NULL, col1_7 INT8 NOT NULL, col1_8 DECIMAL NOT NULL, col1_9 REGNAMESPACE NOT NULL, PRIMARY KEY (col1_6, col1_9, col1_7, col1_3, col1_8, col1_4, col1_1, col1_2, col1_5, col1_0));
        postgres_1    | 	CREATE INDEX ON table1 (col1_1, col1_3 DESC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_2 ASC, col1_7 DESC, col1_1 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_4 DESC, col1_1 ASC, col1_7 ASC, col1_8, col1_5 ASC, col1_3, col1_2 ASC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_8 ASC, col1_6, col1_5, col1_0 ASC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_4, col1_6 ASC, col1_1 DESC, col1_0 DESC, col1_7 ASC, col1_2 ASC, col1_5 ASC, col1_8 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_8, col1_3, col1_7, col1_0);
        postgres_1    | 	CREATE TABLE table2 (col1_6 FLOAT8 NOT NULL, col1_9 REGNAMESPACE NOT NULL, col1_7 INT8 NOT NULL, col1_3 DATE NOT NULL, col1_8 DECIMAL NOT NULL, col1_4 FLOAT8 NOT NULL, col1_1 BIT(32) NOT NULL, col1_2 INTERVAL NOT NULL, col1_5 OID NOT NULL, col1_0 REGPROCEDURE NOT NULL, col2_10 REGTYPE, PRIMARY KEY (col1_6, col1_9, col1_7, col1_3, col1_8, col1_4, col1_1, col1_2, col1_5, col1_0));
        postgres_1    | 	CREATE INDEX ON table2 (col1_2);
        postgres_1    | 	CREATE INDEX ON table2 (col1_3 DESC, col1_8 ASC, col1_2 ASC, col2_10 DESC, col1_0 ASC, col1_5 DESC, col1_4, col1_7 DESC, col1_6 DESC, col1_9);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col1_7 ASC, col1_0, col1_2 DESC, col1_4, col1_1 ASC, col2_10 DESC);
        postgres_1    | 	ALTER TABLE table2 ADD FOREIGN KEY (col1_8, col1_3, col1_7, col2_10) REFERENCES table1 (col1_8, col1_3, col1_7, col1_0);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('tb', 'yccoz', 'gor', 'oskb', 'b', 'l');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('w');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (603.43s)
        test_1        |     --- FAIL: TestCompare/postgres (2.38s)
        test_1        |         compare_test.go:128: SET experimental_enable_enums = true;
        test_1        |             CREATE TABLE table1 (col1_0 REGPROCEDURE NOT NULL, col1_1 BIT(32) NOT NULL, col1_2 INTERVAL NOT NULL, col1_3 DATE NOT NULL, col1_4 FLOAT8 NOT NULL, col1_5 OID NOT NULL, col1_6 FLOAT8 NOT NULL, col1_7 INT8 NOT NULL, col1_8 DECIMAL NOT NULL, col1_9 REGNAMESPACE NOT NULL, PRIMARY KEY (col1_6, col1_9, col1_7, col1_3, col1_8, col1_4, col1_1, col1_2, col1_5, col1_0));
        test_1        |             CREATE INDEX ON table1 (col1_1, col1_3 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_2 ASC, col1_7 DESC, col1_1 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_4 DESC, col1_1 ASC, col1_7 ASC, col1_8, col1_5 ASC, col1_3, col1_2 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_8 ASC, col1_6, col1_5, col1_0 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_4, col1_6 ASC, col1_1 DESC, col1_0 DESC, col1_7 ASC, col1_2 ASC, col1_5 ASC, col1_8 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_8, col1_3, col1_7, col1_0);
        test_1        |             CREATE TABLE table2 (col1_6 FLOAT8 NOT NULL, col1_9 REGNAMESPACE NOT NULL, col1_7 INT8 NOT NULL, col1_3 DATE NOT NULL, col1_8 DECIMAL NOT NULL, col1_4 FLOAT8 NOT NULL, col1_1 BIT(32) NOT NULL, col1_2 INTERVAL NOT NULL, col1_5 OID NOT NULL, col1_0 REGPROCEDURE NOT NULL, col2_10 REGTYPE, PRIMARY KEY (col1_6, col1_9, col1_7, col1_3, col1_8, col1_4, col1_1, col1_2, col1_5, col1_0));
        test_1        |             CREATE INDEX ON table2 (col1_2);
        test_1        |             CREATE INDEX ON table2 (col1_3 DESC, col1_8 ASC, col1_2 ASC, col2_10 DESC, col1_0 ASC, col1_5 DESC, col1_4, col1_7 DESC, col1_6 DESC, col1_9);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col1_7 ASC, col1_0, col1_2 DESC, col1_4, col1_1 ASC, col2_10 DESC);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col1_8, col1_3, col1_7, col2_10) REFERENCES table1 (col1_8, col1_3, col1_7, col1_0);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('tb', 'yccoz', 'gor', 'oskb', 'b', 'l');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('w');
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: unrecognized configuration parameter "experimental_enable_enums" (SQLSTATE 42704)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2045488&tab=buildLog) on [master@c627e3490d30e8ba88f6c7136717a392a054da4e](https://github.com/cockroachdb/cockroach/commits/c627e3490d30e8ba88f6c7136717a392a054da4e):

```
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('ep', 'rrfcv', 'ljlz', 'szbqjl');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('gsvh', 'ldmrmi');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('ifk', 'skq', 'rb', 'jlfil', 'ma');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('ixqxxp');
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: unrecognized configuration parameter "experimental_enable_enums" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (285.67s)
        test_1        |         compare_test.go:160: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		WITH
        test_1        |             			with_44898 (col_250788)
        test_1        |             				AS (SELECT * FROM (VALUES (4216344652::OID), (NULL), (2616606488::OID)) AS tab_167825 (col_250788)),
        test_1        |             			with_44899 (col_250789)
        test_1        |             				AS (
        test_1        |             					SELECT
        test_1        |             						*
        test_1        |             					FROM
        test_1        |             						(
        test_1        |             							VALUES
        test_1        |             								('-66 years -6 mons -709 days -05:40:50.134646'::INTERVAL),
        test_1        |             								('56 years 602 days 19:55:17.281852'::INTERVAL),
        test_1        |             								(
        test_1        |             									date_trunc(substring(st_asgeojson('0101000000000000000000F87F000000000000F87F'::GEOMETRY::GEOMETRY, 2418326950827233528::INT8::INT8, (-6557836296848056752)::INT8::INT8)::TEXT::TEXT, ';'::TEXT::TEXT, e'\x0f'::TEXT::TEXT)::TEXT::TEXT, '15:02:32.323016'::TIME::TIME)::INTERVAL
        test_1        |             								)
        test_1        |             						)
        test_1        |             							AS tab_167826 (col_250789)
        test_1        |             				)
        test_1        |             		SELECT
        test_1        |             			317087932::OID AS col_250790
        test_1        |             		FROM
        test_1        |             			with_44898 AS cte_ref_12825
        test_1        |             		ORDER BY
        test_1        |             			cte_ref_12825.col_250788 ASC,
        test_1        |             			cte_ref_12825.col_250788,
        test_1        |             			cte_ref_12825.col_250788 ASC,
        test_1        |             			cte_ref_12825.col_250788 DESC
        test_1        |             	)
        test_1        |             		AS tab_167827
        test_1        |             ORDER BY
        test_1        |             	col_250790 NULLS FIRST
        test_1        |         compare_test.go:161: cockroach1: ping: set tcp 172.18.0.5:54648: use of closed network connection
        test_1        | FAIL
        compare_cockroach2_1 exited with code 2
        Stopping compare_postgres_1 ... 
        Stopping compare_postgres_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2046710&tab=buildLog) on [master@17c8048e80935f8a01477416980d18bf39cba1bb](https://github.com/cockroachdb/cockroach/commits/17c8048e80935f8a01477416980d18bf39cba1bb):

```
        test_1        |     --- FAIL: TestCompare/mutators (279.15s)
        test_1        |         compare_test.go:160: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		WITH
        test_1        |             			with_42335 (col_260848) AS (SELECT * FROM (VALUES (NULL), (NULL)) AS tab_158215 (col_260848)),
        test_1        |             			with_42336 (col_260849)
        test_1        |             				AS (
        test_1        |             					SELECT
        test_1        |             						*
        test_1        |             					FROM
        test_1        |             						(
        test_1        |             							VALUES
        test_1        |             								(
        test_1        |             									(('{"test": "json"}'::JSONB::JSONB->(-128)::INT8::INT8)::JSONB::JSONB->>NULL::INT8)::TEXT
        test_1        |             								),
        test_1        |             								(NULL),
        test_1        |             								(
        test_1        |             									st_asgeojson('0101000020E6100000000000000000F87F000000000000F87F'::GEOGRAPHY::GEOGRAPHY)::TEXT
        test_1        |             								),
        test_1        |             								(NULL),
        test_1        |             								(NULL)
        test_1        |             						)
        test_1        |             							AS tab_158216 (col_260849)
        test_1        |             				)
        test_1        |             		SELECT
        test_1        |             			cte_ref_12199.col_260848 AS col_260850, cte_ref_12198.col_260848 AS col_260851
        test_1        |             		FROM
        test_1        |             			with_42335 AS cte_ref_12198, with_42335 AS cte_ref_12199
        test_1        |             		ORDER BY
        test_1        |             			cte_ref_12198.col_260848,
        test_1        |             			cte_ref_12199.col_260848,
        test_1        |             			cte_ref_12198.col_260848 DESC,
        test_1        |             			cte_ref_12199.col_260848 ASC,
        test_1        |             			cte_ref_12199.col_260848 DESC
        test_1        |             	)
        test_1        |             		AS tab_158217
        test_1        |             ORDER BY
        test_1        |             	col_260850 NULLS FIRST, col_260851 NULLS FIRST
        test_1        |         compare_test.go:161: cockroach2: ping: set tcp 172.18.0.5:45388: use of closed network connection
        test_1        |     --- FAIL: TestCompare/postgres (0.01s)
        test_1        |         compare_test.go:118: pgx conn: dial tcp 172.18.0.3:26257: connect: connection refused
        test_1        | FAIL
        compare_cockroach2_1 exited with code 2
        Stopping compare_postgres_1 ... 
        Stopping compare_postgres_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2050306&tab=buildLog) on [master@0b65365fce6a9bdde6c611c92031877ade88a8b2](https://github.com/cockroachdb/cockroach/commits/0b65365fce6a9bdde6c611c92031877ade88a8b2):

```
        test_1        |             CREATE TABLE table1 (col1_0 BOOL NOT NULL, col1_1 REGTYPE NOT NULL, col1_2 REGNAMESPACE NOT NULL, col1_3 INT8 NOT NULL, col1_4 TIMESTAMPTZ NOT NULL, col1_5 FLOAT8 NOT NULL, col1_6 INTERVAL NOT NULL, col1_7 DATE NOT NULL, col1_8 INT8 NOT NULL, col1_9 REGPROC NOT NULL, col1_10 REGPROCEDURE NOT NULL, col1_11 VARCHAR NOT NULL, col1_12 INTERVAL NOT NULL, col1_13 REGTYPE NOT NULL, col1_14 FLOAT8 NOT NULL, col1_15 BIT(41) NOT NULL, col1_16 DATE NOT NULL, col1_17 VARCHAR NOT NULL, PRIMARY KEY (col1_3, col1_5, col1_6, col1_7, col1_15, col1_1, col1_10, col1_2, col1_16, col1_11, col1_8, col1_13, col1_4, col1_12, col1_17, col1_9, col1_14, col1_0));
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_13 DESC, col1_6, col1_17, col1_2, col1_15, col1_16 ASC);
        test_1        |             CREATE TABLE table2 (col2_0 INT8 NULL, col2_1 BOOL NULL, col2_2 VARCHAR NULL, col2_3 TIME NOT NULL, col2_4 OID, col2_5 BYTEA NOT NULL, col2_6 BIT(27)[], col2_7 DECIMAL, col2_8 REGTYPE, col2_9 UUID NOT NULL, col2_10 UUID NOT NULL, col2_11 REGTYPE NULL, col2_12 TIME NULL, col2_13 REGNAMESPACE, col2_14 TIMESTAMPTZ, col2_15 TIMESTAMP NOT NULL, col2_16 UUID NOT NULL, col2_17 BIT(49) NULL, PRIMARY KEY (col2_16));
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_3 DESC, col2_14 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_15, col2_9, col2_10, col2_13, col2_8 DESC, col2_14, col2_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_7 DESC, col2_3, col2_1 ASC, col2_4 DESC, col2_0 ASC, col2_5, col2_10 DESC, col2_15, col2_11 ASC, col2_17, col2_14 ASC, col2_13 DESC);
        test_1        |             CREATE INDEX ON table2 (col2_14 DESC) INCLUDE (col2_0, col2_1, col2_2, col2_3, col2_8, col2_13, col2_15, col2_17);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_9 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_17 DESC, col2_9);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_11, col2_1 DESC, col2_7, col2_10, col2_16, col2_12 DESC, col2_3 DESC, col2_4 DESC, col2_15, col2_13) INCLUDE (col2_0, col2_2, col2_5, col2_6, col2_8, col2_9, col2_14, col2_17);
        test_1        |             CREATE INDEX ON table2 (col2_5 ASC, col2_7, col2_8, col2_10 ASC, col2_4, col2_12 ASC, col2_1 DESC, col2_16 DESC, col2_15 ASC, col2_3, col2_9 ASC) INCLUDE (col2_0);
        test_1        |             CREATE TABLE table3 (col3_0 TIMESTAMPTZ, col3_1 BIT(2) NULL, col3_2 INT8 NOT NULL, col3_3 CHAR NULL, col3_4 UUID NULL, col3_5 OID[] NOT NULL, col3_6 OID NOT NULL, col3_7 BIT(39) NOT NULL, col3_8 BYTEA, col3_9 DECIMAL, col3_10 DATE NULL, col3_11 "char", col3_12 NAME NULL, col3_13 FLOAT8 NOT NULL, col3_14 CHAR, col3_15 INTERVAL NULL, PRIMARY KEY (col3_13));
        test_1        |             CREATE INDEX ON table3 (col3_14, col3_4 ASC, col3_6);
        test_1        |             CREATE INDEX ON table3 (col3_7 DESC, col3_11, col3_14, col3_12, col3_4 ASC, col3_6, col3_10 ASC, col3_15 DESC, col3_13 ASC, col3_0, col3_8 DESC, col3_3, col3_2 DESC, col3_9 DESC, col3_1 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_11 DESC, col3_1, col3_4 DESC, col3_7) INCLUDE (col3_3, col3_6, col3_8, col3_9, col3_10, col3_14, col3_15);
        test_1        |             CREATE INDEX ON table3 (col3_8, col3_9, col3_6 ASC, col3_3 ASC, col3_2, col3_13);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_11, col3_0 ASC, col3_15, col3_4 ASC, col3_6, col3_10 ASC, col3_1, col3_2 ASC, col3_12 ASC, col3_9 ASC, col3_8);
        test_1        |             CREATE INDEX ON table3 (col3_2, col3_9 ASC, col3_11, col3_6 ASC);
        test_1        |             CREATE TABLE table4 (col3_13 FLOAT8 NOT NULL, col4_1 FLOAT8 NOT NULL, col4_2 REGNAMESPACE NULL, col4_3 BOOL, col4_4 INT8 NULL, PRIMARY KEY (col3_13, col4_4));
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1);
        test_1        |             CREATE INDEX ON table4 (col4_1 DESC, col4_4, col4_2 ASC, col4_3 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_3 ASC, col3_13);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_3 ASC, col4_4 ASC, col4_1 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_3 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_3, col4_4 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2, col4_1, col4_3);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col3_13 DESC, col4_1 DESC, col4_3, col4_2 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_4, col4_1 ASC, col3_13 ASC, col4_3);
        test_1        |             CREATE TABLE table5 (col3_13 FLOAT8 NOT NULL, col4_4 INT8 NULL, col5_2 BIT(47), col5_3 BIT(13) NULL, col5_4 REGPROC, col5_5 BIT(5) NULL, col5_6 UUID NOT NULL, PRIMARY KEY (col3_13, col4_4));
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_3 ASC, col5_2, col4_4 DESC, col5_4 DESC, col5_6, col5_5 DESC, col3_13 DESC);
        test_1        |             CREATE INDEX ON table5 (col4_4 DESC, col5_6) INCLUDE (col5_4);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_5 ASC, col4_4 DESC, col5_3 DESC) INCLUDE (col5_2, col5_6);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_3 DESC, col5_4);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('iam', 'cz', 'pq', 'nbmsi', 'fhv', 'a');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('lyl');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('ibqr', 'ndq', 'yogw', 'urd');
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: unrecognized configuration parameter "experimental_enable_enums" (SQLSTATE 42704)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    TestComposeCompare: compose_test.go:49: exit status 1
--- FAIL: TestComposeCompare (626.51s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(compose).TestComposeCompare failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2054004&tab=buildLog) on [master@3a03f3843a8cdf04f82c52753c61cf01b0d2ddcd](https://github.com/cockroachdb/cockroach/commits/3a03f3843a8cdf04f82c52753c61cf01b0d2ddcd):

```
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_5 DESC, col4_2 DESC, col4_1 DESC, col4_3 ASC) INCLUDE (col4_6);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0 ASC, col4_1, col4_4);
        test_1        |             CREATE INDEX ON table4 (col4_3, col4_1 DESC, col4_4 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_3, col4_1, col4_2 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_0 ASC, col4_3, col4_1) INCLUDE (col4_6);
        test_1        |             CREATE INDEX ON table4 (col4_5 DESC, col4_4 DESC, col4_2 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_0 DESC, col4_3, col4_4 ASC, col4_2 ASC, col4_5, col4_1 ASC) INCLUDE (col4_6);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col2_9, col2_4) REFERENCES table3 (col3_3, col3_8) ON DELETE CASCADE;
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('v', 'zicnr', 'vkxioo', 'hq', 'bkkxm');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('e', 'm', 'jst');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('aswqf', 'eqz', 'bpz', 'mm');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('ig', 'tvlxhr', 'xhulv', 'jqymw');
        test_1        |             
        test_1        |         compare_test.go:129: postgres: exec: ERROR: unrecognized configuration parameter "experimental_enable_enums" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (234.34s)
        test_1        |         compare_test.go:160: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			count(*) AS col_232392
        test_1        |             		FROM
        test_1        |             			postgres.public.table2 AS tab_156150
        test_1        |             		WHERE
        test_1        |             			true
        test_1        |             		GROUP BY
        test_1        |             			tab_156150.col2_0, tab_156150.col2_1
        test_1        |             		HAVING
        test_1        |             			ilike_escape(st_asgeojson('0101000020E6100000000000000000F87F000000000000F87F'::GEOGRAPHY::GEOGRAPHY, 3744031619755677656::INT8::INT8)::TEXT::TEXT, e'U\x06m\x02M"5M\r'::TEXT::TEXT, 'q'::TEXT::TEXT)::BOOL
        test_1        |             		ORDER BY
        test_1        |             			tab_156150.col2_0 ASC,
        test_1        |             			tab_156150.col2_0 ASC,
        test_1        |             			tab_156150.col2_0,
        test_1        |             			tab_156150.col2_1,
        test_1        |             			tab_156150.col2_1,
        test_1        |             			tab_156150.col2_0 ASC,
        test_1        |             			tab_156150.col2_0 DESC
        test_1        |             	)
        test_1        |             		AS tab_156151
        test_1        |             ORDER BY
        test_1        |             	col_232392 NULLS FIRST
        test_1        |         compare_test.go:161: cockroach1: ping: set tcp 172.18.0.5:52898: use of closed network connection
        test_1        | FAIL
        compare_cockroach1_1 exited with code 2
        Stopping compare_postgres_1 ... 
        Stopping compare_postgres_1 ... done
        Aborting on container exit...
        
    TestComposeCompare: compose_test.go:49: exit status 1
--- FAIL: TestComposeCompare (253.31s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestComposeCompare.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/mutations/mutations.go
+++ b/pkg/sql/mutations/mutations.go
@@ -536,7 +536,7 @@ func postgresMutator(rng *rand.Rand, q string) string {
 var postgresStatementMutator MultiStatementMutation = func(rng *rand.Rand, stmts []tree.Statement) (mutated []tree.Statement, changed bool) {
 	for _, stmt := range stmts {
 		switch stmt := stmt.(type) {
-		case *tree.SetClusterSetting:
+		case *tree.SetClusterSetting, *tree.SetVar:
 			continue
 		case *tree.CreateTable:
 			if stmt.Interleave != nil {

